core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -74,18 +74,20 @@ open class TileInfo {
 
     /** Returns military, civilian and air units in tile */
     fun getUnits(): List<MapUnit> {
-        if(militaryUnit==null && civilianUnit==null && airUnits.isEmpty())
+        if (militaryUnit == null && civilianUnit == null && airUnits.isEmpty())
             return emptyList() // for performance reasons - costs much less to initialize than an empty ArrayList or list()
-        val list = ArrayList<MapUnit>(2)
-        if(militaryUnit!=null) list.add(militaryUnit!!)
-        if(civilianUnit!=null) list.add(civilianUnit!!)
+        val list = ArrayList<MapUnit>(2 + airUnits.size)
+        if (militaryUnit != null) list.add(militaryUnit!!)
+        if (civilianUnit != null) list.add(civilianUnit!!)
         list.addAll(airUnits)
         return list
     }
 
     fun getCity(): CityInfo? = owningCity
 
-    fun getLastTerrain(): Terrain = if (terrainFeature != null) getTerrainFeature()!! else if(naturalWonder != null) getNaturalWonder() else getBaseTerrain()
```

Comment:
This is even worse =|
It wasn't the nicest function to begin with (probably should ave been converted to when) but now it's completely unreadable